test/e2e/upgrade_test.go

```diff
@@ -31,15 +31,70 @@ const (
+func getNodeAffinities() ([]*v1.Affinity, error) {
+	var affinities []*v1.Affinity
+	if gc.numberOfMachineNodes >= 1 {
+		affinityNoByoh, err := getNodeAffinityForLabel(v1.NodeSelectorOpDoesNotExist, controllers.BYOHLabel)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error creating node affinity without BYOH label")
+		}
+		affinities = append(affinities, affinityNoByoh)
+	}
+	if gc.numberOfBYOHNodes >= 1 {
+		affinityByoh, err := getNodeAffinityForLabel(v1.NodeSelectorOpIn, controllers.BYOHLabel, "true")
+		if err != nil {
+			return nil, errors.Wrapf(err, "error creating node affinity for BYOH label")
+		}
+		affinities = append(affinities, affinityByoh)
+	}
+	return affinities, nil
+}
+
+// getLinuxJobCommand creates a Job to curl Windows webserver every 5 seconds.
+// If the webserver is not accessible for few minutes Curl Windows webserver
+// exit the pod with an error.
+func getLinuxJobCommand(clusterIP string) []string {
+	return []string{
+		"bash",
+		"-c",
+		" while true;" +
+			" do curl " + clusterIP + ";" +
+			" if [ $? != 0 ]; then" +
+			" sleep 60;" +
+			" curl " + clusterIP + " || exit 1;" +
+			" fi; " +
+			"sleep 5; " +
+			"done"}
+}
+
 // upgradeTestSuite tests behaviour of the operator when an upgrade takes place.
 func upgradeTestSuite(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
-	// test if Windows workloads are running by creating a Job that curls the workloads continuously.
-	testerJob, err := testCtx.deployWindowsWorkloadAndTester()
-	require.NoError(t, err, "error testing Windows workloads")
-	defer testCtx.deleteJob(testerJob.Name)
+	// generate node affinities
+	affinities, err := getNodeAffinities()
+	require.NoError(t, err, "error getting node affinity")
+	// loop affinities
+	for _, affinity := range affinities {
+		// test if Windows workloads are running by creating a Job that curls the workloads continuously.
+		deployment, err := testCtx.deployWindowsWebServer("win-webserver", affinity)
+		require.NoErrorf(t, err, "error creating Windows Webserver deployment for upgrade test")
+
+		// create a clusterIP service which can be used to reach the Windows webserver
+		intermediarySVC, err := testCtx.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+		require.NoErrorf(t, err, "error creating service for deployment %s", deployment.Name)
+
+		// create a Job object that continuously curls the webserver every 5 seconds.
+		testerJob, err := testCtx.createLinuxJob(windowsWorkloadTesterJob, getLinuxJobCommand(intermediarySVC.Spec.ClusterIP))
+		require.NoErrorf(t, err, "error creating linux job %s", windowsWorkloadTesterJob)
+
+		// stack the deferred calls in reverse to ensure the right execution order (LIFO)
+		defer testCtx.deleteJob(testerJob.Name)
+		defer testCtx.deleteService(intermediarySVC.Name)
+		defer testCtx.deleteDeployment(deployment.Name)
+	}
```

Comment:
I don't see why `deployWindowsWorkloadAndTester()` needed to be removed. My suggestion would be to keep the helper and just add an `affinity *v1.Affinity` parameter to it and leave it as a helper. So changes here would be reduced to:
 
```suggestion
	// generate node affinities
	affinities, err := getNodeAffinities()
	require.NoError(t, err, "error getting node affinity")
	// loop affinities
	for _, affinity := range affinities {
		// test if Windows workloads are running by creating a Job that curls the workloads continuously.
		testerJob, err := testCtx.deployWindowsWorkloadAndTester(affinity)
		require.NoError(t, err, "error testing Windows workloads")
		defer testCtx.deleteJob(testerJob.Name)
	}
```